money.go

```diff
@@ -27,12 +27,26 @@ func defaultUnmarshalJSON(m *Money, b []byte) error {
 	if err != nil {
 		return err
 	}
-	ref := New(int64(data["amount"].(float64)), data["currency"].(string))
+
+	amount := int64(data["amount"].(float64))
+	currency := data["currency"].(string)
+
+	var ref *Money
+	if amount == 0 && currency == "" {
+		ref = &Money{}
+	} else {
+		ref = New(int64(data["amount"].(float64)), data["currency"].(string))
+	}
+
 	*m = *ref
 	return nil
 }
 
 func defaultMarshalJSON(m Money) ([]byte, error) {
+	if m == (Money{}) {
```

Comment:
I was thinking more like

```
func (m *Money) IsZero() bool {
  m == (Money{})
}
```

So that if the fields or anything changes in `Money` it doesn't have to be updated.

Reply:
Sorry for late response. Looks reasonable, let's add `isZero` as you suggest

Reply:
I'm going to skip this, it looks like `IsZero` is already in there with different semantics (it means is the amount zero), what I was thinking was more like `time.Now().IsZero()` meaning "is this a zero value of money".